### PR TITLE
Unittest/arrow deserialization schema

### PR DESCRIPTION
**Pull request by Sridhar-Durgam2022**
Unit test cases for ArrowDeserializationSchema

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -236,6 +237,16 @@
             <artifactId>jsqlparser</artifactId>
             <version>${jsqlparser.version}</version>
         </dependency>
+        <dependency>
+           <groupId>joda-time</groupId>
+           <artifactId>joda-time</artifactId>
+           <version>${joda.version}</version>
+        </dependency>
+        <dependency>
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply pom.xml:248]: Addressed

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]
please rebase it to avoid all those file deletions

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+        instance = null;
+      } finally {
+        instantiated = true;
+      }
+      return instance;
+    }
+
+    public long convertDate(Object object) {
+      final java.util.Date value = (Date) object;
+      return value.getTime();
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:323]: removed joda-time

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+      if (!(object.toString().length() == 26)) {
+        final java.time.LocalDateTime value = (LocalDateTime) object;
+        ZonedDateTime zdt = value.atZone(zoneId);
+        Date output = Date.from(zdt.toInstant());
+        dateTime = output.getTime();
+      } else if (object.toString().length() == 26) {
+        final java.time.LocalDateTime value = (LocalDateTime) object;
+        ZonedDateTime zdt = value.atZone(zoneId);
+        Date output = Date.from(zdt.toInstant());
+        dateTime = output.getTime();
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:346]: removed joda-time

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+      final java.util.Date value = (Date) object;
+      return value.getTime();
+    }
+
+    public int convertTime(Object object) {
+      ZoneId zoneId = ZoneId.of("UTC");
+      final java.time.LocalDateTime value = (LocalDateTime) object;
+      ZonedDateTime zdt = value.atZone(zoneId);
+      Date output = Date.from(zdt.toInstant());
+      return (int) output.getTime();
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:331]: removed joda-time

File: src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+    }
+
+    public long convertTimestamp(Object object) {
+      long dateTime = 0;
+      ZoneId zoneId = ZoneId.of("UTC");
+      if (!(object.toString().length() == 26)) {
+        final java.time.LocalDateTime value = (LocalDateTime) object;
+        ZonedDateTime zdt = value.atZone(zoneId);
+        Date output = Date.from(zdt.toInstant());
+        dateTime = output.getTime();
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/arrow/util/ArrowToRowDataConverters.java:341]: removed joda-time

File: src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+  @Test
+  public void deserializeTest() throws IOException {
+    String jsonStringArrowSchema = "Schema<word: Utf8, word_count: Int(64, true)>";
+    ArrowDeserializationSchema<VectorSchemaRoot> arrowDeserializationSchema =
+        new ArrowDeserializationSchema<VectorSchemaRoot>(
+            VectorSchemaRoot.class, jsonStringArrowSchema, null);
+    assertThat(arrowDeserializationSchema).isNotNull();
+  }
+
+  public ReadSession getReadSession() {
```
Comment (VinayLondhe14) [inline src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java:76]: Please remove this method. Why do we have it here?

File: src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+    Schema schema =
+        MessageSerializer.deserializeSchema(
+            new ReadChannel(
+                new ByteArrayReadableSeekableByteChannel(
+                    arrowSchema.getSerializedSchema().toByteArray())));
+    return schema;
+  }
+
+  @SuppressWarnings("unused")
+  private MockDynamicTableContext createContextObject() {
```
Comment (VinayLondhe14) [inline src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java:127]: Please remove this method. Why do we have it here? It is not used anywhere.

File: src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+
+public class ArrowDeserializationSchemaTest {
+
+  @Test
+  public void deserializeTest() throws IOException {
+    String jsonStringArrowSchema = "Schema<word: Utf8, word_count: Int(64, true)>";
+    ArrowDeserializationSchema<VectorSchemaRoot> arrowDeserializationSchema =
+        new ArrowDeserializationSchema<VectorSchemaRoot>(
+            VectorSchemaRoot.class, jsonStringArrowSchema, null);
+    assertThat(arrowDeserializationSchema).isNotNull();
```
Comment (VinayLondhe14) [inline src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java:73]: Are we only testing that the `arrowDeserializationSchema` is not null? Where are the other assertions and what about testing other methods in the `ArrowDeserializationSchema` class?

File: src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java (hunk 7 above)
Comment (Sridhar-Durgam2022) [reply src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java]: removed

File: src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java (hunk 8 above)
Comment (Sridhar-Durgam2022) [reply src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java]: addressed

File: src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+
+  @Test
+  public void getProducedType() {
+    assertThat(arrowDeserializationSchema.getProducedType()).isEqualTo(null);
+  }
+
+  @Test
+  public void testEquals() {
+
+    assertThat(arrowDeserializationSchema.equals(null)).isEqualTo(false);
```
Comment (davidrabinowitz) [inline src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java:71]: redundant test

File: src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+
+    ArrowDeserializationSchema<VectorSchemaRoot> arrowDeserializationSchemaTest =
+        new ArrowDeserializationSchema<VectorSchemaRoot>(
+            VectorSchemaRoot.class, jsonStringArrowSchema, null);
+    assertThat(arrowDeserializationSchema.equals(arrowDeserializationSchemaTest)).isEqualTo(true);
```
Comment (davidrabinowitz) [inline src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java:76]: how this helps? both objects are generated in the same way.

Also `assertThat(arrowDeserializationSchema).isEqualTo(arrowDeserializationSchemaTest);`

File: src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+  public void testIsEndOfStream() {
+
+    VectorSchemaRoot root = Mockito.mock(VectorSchemaRoot.class);
+    assertThat(arrowDeserializationSchema.isEndOfStream(null)).isEqualTo(true);
+    assertThat(arrowDeserializationSchema.isEndOfStream(root)).isEqualTo(false);
+  }
+
+  @Test
+  public void getProducedType() {
+    assertThat(arrowDeserializationSchema.getProducedType()).isEqualTo(null);
```
Comment (davidrabinowitz) [inline src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java:65]: `ssertThat(arrowDeserializationSchema.getProducedType()).isNull()`

File: src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+        () -> {
+          arrowDeserializationSchema.deserialize(response.toByteArray());
+        });
+  }
+
+  @Test
+  public void testIsEndOfStream() {
+
+    VectorSchemaRoot root = Mockito.mock(VectorSchemaRoot.class);
+    assertThat(arrowDeserializationSchema.isEndOfStream(null)).isEqualTo(true);
```
Comment (davidrabinowitz) [inline src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java:59]: assertThat(arrowDeserializationSchema.isEndOfStream(null)).isTrue();

File: src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+    assertThat(arrowDeserializationSchema.isEndOfStream(root)).isEqualTo(false);
```
Comment (davidrabinowitz) [inline src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java:60]: assertThat(arrowDeserializationSchema.isEndOfStream(root)).isFalse();

File: src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+    jsonStringArrowSchema = "Schema<word: Utf8, word_count: Int(64, true)>";
+    arrowDeserializationSchema =
+        new ArrowDeserializationSchema<VectorSchemaRoot>(
+            VectorSchemaRoot.class, jsonStringArrowSchema, null);
+  }
+
+  @Test
+  public void deserializeTest() throws IOException {
+
+    assertThat(arrowDeserializationSchema).isNotNull();
```
Comment (davidrabinowitz) [inline src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java:45]: redundant check

File: src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+    ReadRowsResponse response = Mockito.mock(ReadRowsResponse.class);
+    assertThrows(
```
Comment (davidrabinowitz) [inline src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java:47]: This test is meaningless - We should test an actual response

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]

File: src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java (hunk 14 above)
Comment (Sridhar-Durgam2022) [reply src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java]: Changed

File: src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java (hunk 10 above)
Comment (Sridhar-Durgam2022) [reply src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java]: Removed as not necessary

File: src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java (hunk 11 above)
Comment (Sridhar-Durgam2022) [reply src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java]: changed

File: src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java (hunk 9 above)
Comment (Sridhar-Durgam2022) [reply src/test/java/com/google/cloud/flink/bigquery/ArrowDeserializationSchemaTest.java]: removed

**Pull request closed without merging**